ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceBlobsSidecarAvailabilityCheckerTest.java

```diff
@@ -0,0 +1,140 @@
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecVersion;
+import tech.pegasys.teku.spec.TestSpecFactory;
+import tech.pegasys.teku.spec.datastructures.execution.versions.eip4844.BlobsSidecar;
+import tech.pegasys.teku.spec.logic.common.helpers.MiscHelpers;
+import tech.pegasys.teku.spec.logic.versions.eip4844.blobs.BlobsSidecarAvailabilityChecker;
+import tech.pegasys.teku.spec.logic.versions.eip4844.blobs.BlobsSidecarAvailabilityChecker.BlobsSidecarValidationResult;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+import tech.pegasys.teku.storage.client.RecentChainData;
+
+public class ForkChoiceBlobsSidecarAvailabilityCheckerTest {
+  private final Spec spec = TestSpecFactory.createMainnetEip4844();
+  private final SpecVersion specVersionMock = mock(SpecVersion.class);
+  private final MiscHelpers miscHelpers = mock(MiscHelpers.class);
+  private final DataStructureUtil dataStructureUtil = new DataStructureUtil(spec);
+  private final RecentChainData recentChainData = mock(RecentChainData.class);
+
+  private final long availabilityWindow =
+      (long) spec.getGenesisSpec().getSlotsPerEpoch() * MIN_EPOCHS_FOR_BLOBS_SIDECARS_REQUESTS;
+
+  private BlobsSidecar blobsSidecar;
+  private BlobsSidecarAvailabilityChecker blobsSidecarAvailabilityChecker;
+
+  @BeforeEach
+  void setUp() {
+    when(specVersionMock.getSlotsPerEpoch()).thenReturn(spec.getGenesisSpec().getSlotsPerEpoch());
+    when(specVersionMock.miscHelpers()).thenReturn(miscHelpers);
+  }
+
+  @Test
+  void shouldReturnNotRequired() {
+    prepareBlockAndBlobOutsideAvailabilityWindow(false);
+
+    blobsSidecarAvailabilityChecker.initiateDataAvailabilityCheck();
+    assertThat(blobsSidecarAvailabilityChecker.getAvailabilityCheckResult())
+        .isCompletedWithValueMatching(
+            result ->
+                !result.isFailure()
+                    && result.isNotRequired()
+                    && result.getBlobsSidecar().isEmpty());
+  }
+
+  @Test
+  void shouldReturnNotAvailable() {
+    prepareBlockAndBlobInAvailabilityWindow(false);
+
+    blobsSidecarAvailabilityChecker.initiateDataAvailabilityCheck();
+    assertThat(blobsSidecarAvailabilityChecker.getAvailabilityCheckResult())
+        .isCompletedWithValueMatching(
+            result ->
+                result.isFailure()
+                    && result.getValidationResult() == BlobsSidecarValidationResult.NOT_AVAILABLE
+                    && result.getBlobsSidecar().isEmpty());
+  }
+
+  @Test
+  void shouldReturnInvalid() {
+    prepareBlockAndBlobInAvailabilityWindow(true);
+
+    blobsSidecarAvailabilityChecker.initiateDataAvailabilityCheck();
+    assertThat(blobsSidecarAvailabilityChecker.getAvailabilityCheckResult())
+        .isCompletedWithValueMatching(
+            result ->
+                result.isFailure()
```

Comment:
I'd add valid case too, when reading this one it's especially unclear why it's invalid until checking whole logic

Reply:
added